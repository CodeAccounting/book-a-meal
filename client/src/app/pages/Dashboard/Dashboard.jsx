import React, { Component, Fragment } from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';
import CatererView from '../../shared/CatererView';
import { OrderHistoryTable } from '../../shared/Tables';
import { userPropTypes, catererOrdersObjPropTypes } from '../../../helpers/proptypes';

/**
 * @exports
 * @class Dashboard
 * @extends Component
 * @classdesc Creates Dashboard Component
 * @returns {JSX} Dashboard Component
 */
class Dashboard extends Component {
  static propTypes = {
    ...userPropTypes,
    ...catererOrdersObjPropTypes,
    isFetching: PropTypes.bool.isRequired,
    logout: PropTypes.func.isRequired,
    fetchOrders: PropTypes.func.isRequired,
    deliverOrder: PropTypes.func.isRequired,
  }

  /**
   * @memberof Dashboard
   * @returns {JSX} Dashboard Component
   */
  componentDidMount() {
    const todaysDate = moment().format('YYYY-MM-DD');
    this.props.fetchOrders('caterer', todaysDate);
  }

  /**
   * @memberof Dashboard
   * @returns {JSX} Board Component
   */
  renderDashBoard = () => (
    <Fragment>
      <div className="date d-none-md">
        <h2>{moment().format('dddd[,] Do MMMM YYYY')}&nbsp; &nbsp; {moment().format('HH:mm')}</h2>
      </div>
      <div className="card-group summary">
        <div className="card total-sum">
          <div className="count">{this.props.orders.length}</div>
          <div>{'Today\'s Orders'}</div>
        </div>
        <div className="card pending">
          <div className="count">{this.props.pendingOrders}</div>
          <div>Pending Orders</div>
        </div>
        <div className="card total-cash">
          <div className="count">&#8358; {this.props.totalCashEarned}</div>
          <div>{'Today\'s Revenue'}</div>
        </div>
      </div>
      <div className="order-details">
        <OrderHistoryTable
          orders={this.props.orders}
          deliverOrder={this.props.deliverOrder}
        />
      </div>
    </Fragment>
  )


  /**
   * @memberof Dashboard
   * @returns {JSX} Dashboard Component
   */
  render() {
    const { user, logout, isFetching } = this.props;

    return (
      <CatererView user={user} logout={logout} type="dashboard" isFetching={isFetching}>
        {this.renderDashBoard()}
      </CatererView>
    );
  }
}

export default Dashboard;
