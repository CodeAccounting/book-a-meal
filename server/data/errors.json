{
  "401": "Unauthorized",
  "403": "Forbidden",
  "404": "Not Found",
  "422": "Validation Error",
  "500": "Internal Server Error",
  "empty": "Empty PUT Requests Are Not Allowed"
}