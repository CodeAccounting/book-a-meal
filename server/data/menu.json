[
  {
    "userId": "8356954a-9a42-4616-8079-887a73455a7f",
     "menuId": "a9fa6cb3-9f5e-46fa-b641-388f898ca824",
     "date": "2018-05-06",
     "meals": [
        "81211c24-51c0-46ec-b1e0-18db55880958",
        "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
        "baa0412a-d167-4d2b-b1d8-404cb8f02631",
        "f9eb7652-125a-4bcb-ad81-02f84901cdc3"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  },
  {
    "userId": "8356954a-9a42-4616-8079-887a73455a7f",
     "menuId": "1adfcfe7-c66d-42d2-82fd-39c1decd290a",
     "date": "2018-04-06",
     "meals": [
        "a3c35e8f-da7a-4113-aa01-a9c0fc088539",
        "3ec802c6-0a32-4d29-b27b-42e0f4b532dd"
     ],
     "created": "2018-04-06T14:34:90.000",
     "updated": "2018-04-06T14:34:90.000"
  },
  {
    "userId": "ac1b253c-6b33-439b-ab6f-805a4fdd2e05",
     "menuId": "3a4fb034-6e52-4bde-9abc-91119a39478d",
     "date": "2018-05-07",
     "meals": [
        "72a3417e-45c8-4559-8b74-8b5a61be8614"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  },
  {
    "userId": "ac1b253c-6b33-439b-ab6f-805a4fdd2e05",
     "menuId": "f43f3d49-a6c9-476d-b65a-1af772ff0f36",
     "date": "2018-04-06",
     "meals": [
        "72a3417e-45c8-4559-8b74-8b5a61be8614",
        "46ced7aa-eed5-4462-b2c0-153f31589bdd"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  },
  {
     "userId": "8356954a-9a42-4616-8079-887a73455a7f",
     "menuId": "15421f7a-0f82-4802-b215-e0e8efb6bfb3",
     "date": "2018-04-27",
     "meals": [
        "91b6e41c-0972-4ac5-86da-4ac1f5226e83",
        "f9eb7652-125a-4bcb-ad81-02f84901cdc3",
        "8a65538d-f862-420e-bcdc-80743df06578"
     ],
     "created": "2018-05-06T14:34:90.000",
     "updated": "2018-05-06T14:34:90.000"
  }
]