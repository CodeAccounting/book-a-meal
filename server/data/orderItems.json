[
  {
     "id": 1,
     "mealId": "46ced7aa-eed5-4462-b2c0-153f31589bdd",
     "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
     "quantity": 2,
     "createdAt": "2018-04-06T14:34:90.000",
     "updatedAt": "2018-04-06T14:34:90.000"
  },
  {
     "id": 2,
     "mealId": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
     "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
     "quantity": 1,
     "createdAt": "2018-04-06T14:34:90.000",
     "updatedAt": "2018-04-06T14:34:90.000"
  }, 
  {
    "id": 3,
    "mealId": "72a3417e-45c8-4559-8b74-8b5a61be8614",
    "orderId": "fb097bde-5959-45ff-8e21-51184fa60c25",
    "quantity": 2,
    "createdAt": "2018-04-06T14:34:90.000",
    "updatedAt": "2018-04-06T14:34:90.000"
  },
  {
    "id": 4,
    "mealId": "a3c35e8f-da7a-4113-aa01-a9c0fc088539",
    "orderId": "6ed0963f-9663-4fe2-8ad4-2f06c6294482",
    "quantity": 2,
    "createdAt": "2018-05-06T14:34:90.000",
    "updatedAt": "2018-05-06T14:34:90.000"
  },
  {
    "id": 5,
    "mealId": "baa0412a-d167-4d2b-b1d8-404cb8f02631",
    "orderId": "6ed0963f-9663-4fe2-8ad4-2f06c6294482",
    "quantity": 2,
    "createdAt": "2018-05-06T14:34:90.000",
    "updatedAt": "2018-05-06T14:34:90.000"
  }
]