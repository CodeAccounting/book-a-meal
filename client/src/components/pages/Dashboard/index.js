import { connect, } from 'react-redux';
import { bindActionCreators } from 'redux';
import Dashboard from './Dashboard';
import { fetchOrders, deliverOrder } from '../../../store/operations/orders';
import { logout } from '../../../store/operations/auth';

const mapStateToProps = state => ({
  isFetching: state.isFetching,
  orders: state.orders.items,
  pendingOrders: state.orders.pendingOrders,
  totalCashEarned: state.orders.totalCashEarned
});

const mapDispatchToProps = dispatch => bindActionCreators({
  fetchOrders, deliverOrder, logout
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(Dashboard);
