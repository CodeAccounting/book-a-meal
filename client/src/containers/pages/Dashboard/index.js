import { connect } from 'react-redux';
import Dashboard from '../../../components/pages/Dashboard';
import { fetchOrders, deliverOrder } from '../../../actions/orders';
import { logout } from '../../../actions/auth';

const mapStateToProps = state => ({
  isFetching: state.isFetching,
  orders: state.orders.items,
  pendingOrders: state.orders.pendingOrders,
  totalCashEarned: state.orders.totalCashEarned
});

export default connect(mapStateToProps, { fetchOrders, deliverOrder, logout })(Dashboard);
