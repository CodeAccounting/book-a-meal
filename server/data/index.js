import users from './users.json';
import meals from './meals.json';
import menu from './menu.json';
import orders from './orders.json';
import errors from './errors.json';

export default {
  users, meals, menu, orders, errors
};
