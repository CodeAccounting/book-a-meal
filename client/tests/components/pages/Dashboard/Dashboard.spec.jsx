import React from 'react';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Dashboard from '../../../../src/components/pages/Dashboard/Dashboard';
import ConnectedDashboard from '../../../../src/components/pages/Dashboard';
import { caterer, caterersOrdersObj, initialValues } from '../../../setup/data';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore(initialValues);
const { now } = Date;


describe('Dashboard', () => {
  beforeAll(() => {
    Date.now = jest.fn(() => 0);
  });

  afterAll(() => {
    Date.now = now;
  });

  it('renders correctly when not fetching', () => {
    const shallowWrapper = shallow(<Dashboard
      user={caterer}
      logout={jest.fn()}
      fetchOrders={jest.fn()}
      deliverOrder={jest.fn()}
      {...caterersOrdersObj}
      isFetching={false}
    />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('renders Preloader when fetching', () => {
    const shallowWrapper = shallow(<Dashboard
      user={caterer}
      logout={jest.fn()}
      fetchOrders={jest.fn()}
      deliverOrder={jest.fn()}
      {...caterersOrdersObj}
      isFetching
    />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
    expect(shallowWrapper.find('Preloader')).toBeTruthy();
  });

  it('renders connected component', () => {
    const dispatchMock = jest.fn();
    const comp = (
      <Provider store={store}>
        <ConnectedDashboard
          user={caterer}
          dispatch={dispatchMock}
          {...caterersOrdersObj}
          isFetching={false}
        />
      </Provider>);

    const wrapper = mount(comp);

    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
