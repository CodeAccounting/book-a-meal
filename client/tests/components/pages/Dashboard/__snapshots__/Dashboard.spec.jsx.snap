// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders Preloader when fetching 1`] = `
<CatererView
  isFetching={true}
  logout={[MockFunction]}
  showTime={true}
  type="dashboard"
  user={
    Object {
      "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
      "businessName": "Food Circle",
      "businessPhoneNo": "+2348166557788",
      "email": "food@circle.com",
      "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
      "role": "caterer",
    }
  }
>
  <React.Fragment>
    <div
      className="date d-none-md"
    >
      <h2>
        Thursday, 1st January 1970
            
        00:00
      </h2>
    </div>
    <div
      className="card-group summary"
    >
      <div
        className="card total-sum"
      >
        <div
          className="count"
        >
          4
        </div>
        <div>
          Today's Orders
        </div>
      </div>
      <div
        className="card pending"
      >
        <div
          className="count"
        >
          0
        </div>
        <div>
          Pending Orders
        </div>
      </div>
      <div
        className="card total-cash"
      >
        <div
          className="count"
        >
          ₦ 
          9200
        </div>
        <div>
          Today's Revenue
        </div>
      </div>
    </div>
    <div
      className="order-details"
    >
      <OrderHistoryTable
        deliverOrder={[MockFunction]}
        orders={
          Array [
            Object {
              "createdAt": "2018-04-06T00:47:03.687Z",
              "customer": Object {
                "email": "emiola@olasanmi.com",
                "firstname": "Emiola",
                "lastname": "Olasanmi",
              },
              "deliveryAddress": "4, Church Street, Yaba",
              "deliveryPhoneNo": "+2348134567890",
              "id": "fb097bde-5959-45ff-8e21-51184fa60c25",
              "meals": Array [
                Object {
                  "delivered": true,
                  "description": "Sharwama contains no animal products, perfect for dieters",
                  "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
                  "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
                  "price": "1200.00",
                  "quantity": 1,
                  "title": "Vegetable Sharwama and Guava Smoothie",
                  "vegetarian": true,
                },
              ],
              "status": "delivered",
              "updatedAt": "2018-04-06T00:47:03.687Z",
            },
            Object {
              "createdAt": "2018-05-01T00:47:03.687Z",
              "customer": Object {
                "email": "olisa@emodi.com",
                "firstname": "Olisa",
                "lastname": "Emodi",
              },
              "deliveryAddress": "4, Church Street, Yaba",
              "deliveryPhoneNo": "+2348134567890",
              "id": "ce228787-f939-40a0-bfd3-6607ca8d2e53",
              "meals": Array [
                Object {
                  "delivered": true,
                  "description": "Meal contains 2 pieces of beef and other assorted meat and fish products",
                  "id": "baa0412a-d167-4d2b-b1d8-404cb8f02631",
                  "imageURL": "https://static.pulse.ng/img/incoming/origs7167742/5270485143-w980-h640/Pounded-yam-and-Egusi-soup.jpg",
                  "price": "2000.00",
                  "quantity": 2,
                  "title": "Semo/Wheat and Egusi Soup",
                  "vegetarian": false,
                },
                Object {
                  "delivered": true,
                  "description": "Meal contains 2 pieces of beef and other assorted meat and fish products",
                  "id": "a3c35e8f-da7a-4113-aa01-a9c0fc088539",
                  "imageURL": "http://www.kalakutahgrills.com/wp-content/uploads/2017/02/my-pics-group-3-007.jpg",
                  "price": "2000.00",
                  "quantity": 2,
                  "title": "Eba and Ogbono Soup",
                  "vegetarian": false,
                },
              ],
              "status": "delivered",
              "updatedAt": "2018-05-01T00:47:03.687Z",
            },
            Object {
              "createdAt": "2018-05-29T00:47:03.687Z",
              "customer": Object {
                "email": "emiola@olasanmi.com",
                "firstname": "Emiola",
                "lastname": "Olasanmi",
              },
              "deliveryAddress": "4, Church Street, Yaba",
              "deliveryPhoneNo": "+2348134567890",
              "id": "fb097bde-5959-45ff-8e21-51184fa60c26",
              "meals": Array [
                Object {
                  "delivered": false,
                  "description": "Sharwama contains no animal products, perfect for dieters",
                  "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
                  "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
                  "price": "1200.00",
                  "quantity": 1,
                  "title": "Vegetable Sharwama and Guava Smoothie",
                  "vegetarian": true,
                },
              ],
              "status": "pending",
              "updatedAt": "2018-05-29T00:47:03.687Z",
            },
            Object {
              "createdAt": "2018-05-27T00:47:03.687Z",
              "customer": Object {
                "email": "emiola@olasanmi.com",
                "firstname": "Emiola",
                "lastname": "Olasanmi",
              },
              "deliveryAddress": "4, Church Street, Yaba",
              "deliveryPhoneNo": "+2348134567890",
              "id": "fb097bde-5959-45ff-8e21-51184fa60c35",
              "meals": Array [
                Object {
                  "delivered": false,
                  "description": "Sharwama contains no animal products, perfect for dieters",
                  "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
                  "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
                  "price": "1200.00",
                  "quantity": 1,
                  "title": "Vegetable Sharwama and Guava Smoothie",
                  "vegetarian": true,
                },
              ],
              "status": "canceled",
              "updatedAt": "2018-05-27T00:47:03.687Z",
            },
          ]
        }
      />
    </div>
  </React.Fragment>
</CatererView>
`;

exports[`Dashboard renders connected component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Dashboard)
    dispatch={[MockFunction]}
    isFetching={false}
    orders={
      Array [
        Object {
          "createdAt": "2018-04-06T00:47:03.687Z",
          "customer": Object {
            "email": "emiola@olasanmi.com",
            "firstname": "Emiola",
            "lastname": "Olasanmi",
          },
          "deliveryAddress": "4, Church Street, Yaba",
          "deliveryPhoneNo": "+2348134567890",
          "id": "fb097bde-5959-45ff-8e21-51184fa60c25",
          "meals": Array [
            Object {
              "delivered": true,
              "description": "Sharwama contains no animal products, perfect for dieters",
              "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
              "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
              "price": "1200.00",
              "quantity": 1,
              "title": "Vegetable Sharwama and Guava Smoothie",
              "vegetarian": true,
            },
          ],
          "status": "delivered",
          "updatedAt": "2018-04-06T00:47:03.687Z",
        },
        Object {
          "createdAt": "2018-05-01T00:47:03.687Z",
          "customer": Object {
            "email": "olisa@emodi.com",
            "firstname": "Olisa",
            "lastname": "Emodi",
          },
          "deliveryAddress": "4, Church Street, Yaba",
          "deliveryPhoneNo": "+2348134567890",
          "id": "ce228787-f939-40a0-bfd3-6607ca8d2e53",
          "meals": Array [
            Object {
              "delivered": true,
              "description": "Meal contains 2 pieces of beef and other assorted meat and fish products",
              "id": "baa0412a-d167-4d2b-b1d8-404cb8f02631",
              "imageURL": "https://static.pulse.ng/img/incoming/origs7167742/5270485143-w980-h640/Pounded-yam-and-Egusi-soup.jpg",
              "price": "2000.00",
              "quantity": 2,
              "title": "Semo/Wheat and Egusi Soup",
              "vegetarian": false,
            },
            Object {
              "delivered": true,
              "description": "Meal contains 2 pieces of beef and other assorted meat and fish products",
              "id": "a3c35e8f-da7a-4113-aa01-a9c0fc088539",
              "imageURL": "http://www.kalakutahgrills.com/wp-content/uploads/2017/02/my-pics-group-3-007.jpg",
              "price": "2000.00",
              "quantity": 2,
              "title": "Eba and Ogbono Soup",
              "vegetarian": false,
            },
          ],
          "status": "delivered",
          "updatedAt": "2018-05-01T00:47:03.687Z",
        },
        Object {
          "createdAt": "2018-05-29T00:47:03.687Z",
          "customer": Object {
            "email": "emiola@olasanmi.com",
            "firstname": "Emiola",
            "lastname": "Olasanmi",
          },
          "deliveryAddress": "4, Church Street, Yaba",
          "deliveryPhoneNo": "+2348134567890",
          "id": "fb097bde-5959-45ff-8e21-51184fa60c26",
          "meals": Array [
            Object {
              "delivered": false,
              "description": "Sharwama contains no animal products, perfect for dieters",
              "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
              "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
              "price": "1200.00",
              "quantity": 1,
              "title": "Vegetable Sharwama and Guava Smoothie",
              "vegetarian": true,
            },
          ],
          "status": "pending",
          "updatedAt": "2018-05-29T00:47:03.687Z",
        },
        Object {
          "createdAt": "2018-05-27T00:47:03.687Z",
          "customer": Object {
            "email": "emiola@olasanmi.com",
            "firstname": "Emiola",
            "lastname": "Olasanmi",
          },
          "deliveryAddress": "4, Church Street, Yaba",
          "deliveryPhoneNo": "+2348134567890",
          "id": "fb097bde-5959-45ff-8e21-51184fa60c35",
          "meals": Array [
            Object {
              "delivered": false,
              "description": "Sharwama contains no animal products, perfect for dieters",
              "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
              "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
              "price": "1200.00",
              "quantity": 1,
              "title": "Vegetable Sharwama and Guava Smoothie",
              "vegetarian": true,
            },
          ],
          "status": "canceled",
          "updatedAt": "2018-05-27T00:47:03.687Z",
        },
      ]
    }
    pendingOrders={0}
    totalCashEarned={9200}
    user={
      Object {
        "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
        "businessName": "Food Circle",
        "businessPhoneNo": "+2348166557788",
        "email": "food@circle.com",
        "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
        "role": "caterer",
      }
    }
  >
    <Dashboard
      deliverOrder={[Function]}
      dispatch={[MockFunction]}
      fetchOrders={[Function]}
      isFetching={false}
      logout={[Function]}
      orders={Array []}
      pendingOrders={0}
      totalCashEarned={0}
      user={
        Object {
          "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
          "businessName": "Food Circle",
          "businessPhoneNo": "+2348166557788",
          "email": "food@circle.com",
          "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
          "role": "caterer",
        }
      }
    >
      <CatererView
        isFetching={false}
        logout={[Function]}
        showTime={true}
        type="dashboard"
        user={
          Object {
            "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
            "businessName": "Food Circle",
            "businessPhoneNo": "+2348166557788",
            "email": "food@circle.com",
            "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
            "role": "caterer",
          }
        }
      >
        <div
          className="admin"
        >
          <Header
            dateType="dashboard"
            showTime={true}
            type="caterer"
          >
            <div
              className="header"
            >
              <div
                className="navbar"
              >
                <Connect(CatererHeader)
                  dateType="dashboard"
                  showTime={true}
                >
                  <CatererHeader
                    currentDay="1970-01-01"
                    dateType="dashboard"
                    dispatch={[Function]}
                    showTime={true}
                  >
                    <div
                      className="menu-control d-none-md"
                    >
                      <a
                        className="d-none-md"
                        href="#sidenav"
                      >
                        <img
                          alt="toggler"
                          src="/img/toggler.png"
                        />
                      </a>
                    </div>
                    <div
                      className="page-title d-flex-md"
                    >
                      <div
                        className="date"
                      >
                        <h2>
                          Thursday, 1st January 1970
                              
                          00:00
                        </h2>
                      </div>
                    </div>
                  </CatererHeader>
                </Connect(CatererHeader)>
              </div>
            </div>
          </Header>
          <div
            className="content"
          >
            <SideNav
              active="dashboard"
              logout={[Function]}
              user={
                Object {
                  "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
                  "businessName": "Food Circle",
                  "businessPhoneNo": "+2348166557788",
                  "email": "food@circle.com",
                  "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                  "role": "caterer",
                }
              }
            >
              <div
                className="sidenav"
                id="sidenav"
              >
                <div
                  className="sidenav-content"
                >
                  <div
                    className="sidenav-header"
                  >
                    <div
                      className="close d-none-md"
                    >
                      <a
                        href="#"
                      >
                        ×
                      </a>
                       
                    </div>
                    <div
                      className="sidenav-title"
                    >
                      <h3>
                        <a
                          href="/"
                        >
                          BOOK-A-MEAL
                        </a>
                      </h3>
                      <div
                        className="username-circle"
                      >
                        <p>
                          F
                        </p>
                      </div>
                      <p>
                        Food Circle
                      </p>
                      <span>
                        <Link
                          className={null}
                          clickHandler={[Function]}
                          id={null}
                        >
                          <button
                            className="link-btn null"
                            id={null}
                            onClick={[Function]}
                          >
                            Logout
                          </button>
                        </Link>
                      </span>
                    </div>
                  </div>
                  <div
                    className="sidenav-body"
                  >
                    <a
                      className="menu-item active"
                      href="/"
                    >
                      Dashboard
                    </a>
                    <a
                      className="menu-item false"
                      href="/meals"
                    >
                      Meals
                    </a>
                    <a
                      className="menu-item false"
                      href="/menu"
                    >
                      Menu
                    </a>
                    <a
                      className="menu-item false"
                      href="/orders"
                    >
                      Orders
                    </a>
                  </div>
                </div>
              </div>
            </SideNav>
            <div
              className="content-wrapper dashboard"
              id="has-modal"
            >
              <div
                className="date d-none-md"
              >
                <h2>
                  Thursday, 1st January 1970
                      
                  00:00
                </h2>
              </div>
              <div
                className="card-group summary"
              >
                <div
                  className="card total-sum"
                >
                  <div
                    className="count"
                  >
                    0
                  </div>
                  <div>
                    Today's Orders
                  </div>
                </div>
                <div
                  className="card pending"
                >
                  <div
                    className="count"
                  >
                    0
                  </div>
                  <div>
                    Pending Orders
                  </div>
                </div>
                <div
                  className="card total-cash"
                >
                  <div
                    className="count"
                  >
                    ₦ 
                    0
                  </div>
                  <div>
                    Today's Revenue
                  </div>
                </div>
              </div>
              <div
                className="order-details"
              >
                <OrderHistoryTable
                  deliverOrder={[Function]}
                  orders={Array []}
                >
                  <div
                    className="table-wrapper"
                  >
                    <table>
                      <thead>
                        <tr>
                          <th>
                            ID
                          </th>
                          <th>
                            Meal Item(s)
                          </th>
                          <th>
                            Customer
                          </th>
                          <th>
                            Amount
                          </th>
                          <th>
                            Date
                          </th>
                          <th>
                            Status
                          </th>
                        </tr>
                      </thead>
                      <tbody />
                    </table>
                  </div>
                </OrderHistoryTable>
              </div>
            </div>
          </div>
          <Footer>
            <div
              className="footer"
            >
              <p>
                © 2018. Iveren Shaguy. All images were gotten from Google and are not mine.
              </p>
            </div>
          </Footer>
        </div>
      </CatererView>
    </Dashboard>
  </Connect(Dashboard)>
</Provider>
`;

exports[`Dashboard renders correctly when not fetching 1`] = `
<CatererView
  isFetching={false}
  logout={[MockFunction]}
  showTime={true}
  type="dashboard"
  user={
    Object {
      "businessAddress": "$, Herbert Macaulay Way, Yaba, Lagos",
      "businessName": "Food Circle",
      "businessPhoneNo": "+2348166557788",
      "email": "food@circle.com",
      "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
      "role": "caterer",
    }
  }
>
  <React.Fragment>
    <div
      className="date d-none-md"
    >
      <h2>
        Thursday, 1st January 1970
            
        00:00
      </h2>
    </div>
    <div
      className="card-group summary"
    >
      <div
        className="card total-sum"
      >
        <div
          className="count"
        >
          4
        </div>
        <div>
          Today's Orders
        </div>
      </div>
      <div
        className="card pending"
      >
        <div
          className="count"
        >
          0
        </div>
        <div>
          Pending Orders
        </div>
      </div>
      <div
        className="card total-cash"
      >
        <div
          className="count"
        >
          ₦ 
          9200
        </div>
        <div>
          Today's Revenue
        </div>
      </div>
    </div>
    <div
      className="order-details"
    >
      <OrderHistoryTable
        deliverOrder={[MockFunction]}
        orders={
          Array [
            Object {
              "createdAt": "2018-04-06T00:47:03.687Z",
              "customer": Object {
                "email": "emiola@olasanmi.com",
                "firstname": "Emiola",
                "lastname": "Olasanmi",
              },
              "deliveryAddress": "4, Church Street, Yaba",
              "deliveryPhoneNo": "+2348134567890",
              "id": "fb097bde-5959-45ff-8e21-51184fa60c25",
              "meals": Array [
                Object {
                  "delivered": true,
                  "description": "Sharwama contains no animal products, perfect for dieters",
                  "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
                  "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
                  "price": "1200.00",
                  "quantity": 1,
                  "title": "Vegetable Sharwama and Guava Smoothie",
                  "vegetarian": true,
                },
              ],
              "status": "delivered",
              "updatedAt": "2018-04-06T00:47:03.687Z",
            },
            Object {
              "createdAt": "2018-05-01T00:47:03.687Z",
              "customer": Object {
                "email": "olisa@emodi.com",
                "firstname": "Olisa",
                "lastname": "Emodi",
              },
              "deliveryAddress": "4, Church Street, Yaba",
              "deliveryPhoneNo": "+2348134567890",
              "id": "ce228787-f939-40a0-bfd3-6607ca8d2e53",
              "meals": Array [
                Object {
                  "delivered": true,
                  "description": "Meal contains 2 pieces of beef and other assorted meat and fish products",
                  "id": "baa0412a-d167-4d2b-b1d8-404cb8f02631",
                  "imageURL": "https://static.pulse.ng/img/incoming/origs7167742/5270485143-w980-h640/Pounded-yam-and-Egusi-soup.jpg",
                  "price": "2000.00",
                  "quantity": 2,
                  "title": "Semo/Wheat and Egusi Soup",
                  "vegetarian": false,
                },
                Object {
                  "delivered": true,
                  "description": "Meal contains 2 pieces of beef and other assorted meat and fish products",
                  "id": "a3c35e8f-da7a-4113-aa01-a9c0fc088539",
                  "imageURL": "http://www.kalakutahgrills.com/wp-content/uploads/2017/02/my-pics-group-3-007.jpg",
                  "price": "2000.00",
                  "quantity": 2,
                  "title": "Eba and Ogbono Soup",
                  "vegetarian": false,
                },
              ],
              "status": "delivered",
              "updatedAt": "2018-05-01T00:47:03.687Z",
            },
            Object {
              "createdAt": "2018-05-29T00:47:03.687Z",
              "customer": Object {
                "email": "emiola@olasanmi.com",
                "firstname": "Emiola",
                "lastname": "Olasanmi",
              },
              "deliveryAddress": "4, Church Street, Yaba",
              "deliveryPhoneNo": "+2348134567890",
              "id": "fb097bde-5959-45ff-8e21-51184fa60c26",
              "meals": Array [
                Object {
                  "delivered": false,
                  "description": "Sharwama contains no animal products, perfect for dieters",
                  "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
                  "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
                  "price": "1200.00",
                  "quantity": 1,
                  "title": "Vegetable Sharwama and Guava Smoothie",
                  "vegetarian": true,
                },
              ],
              "status": "pending",
              "updatedAt": "2018-05-29T00:47:03.687Z",
            },
            Object {
              "createdAt": "2018-05-27T00:47:03.687Z",
              "customer": Object {
                "email": "emiola@olasanmi.com",
                "firstname": "Emiola",
                "lastname": "Olasanmi",
              },
              "deliveryAddress": "4, Church Street, Yaba",
              "deliveryPhoneNo": "+2348134567890",
              "id": "fb097bde-5959-45ff-8e21-51184fa60c35",
              "meals": Array [
                Object {
                  "delivered": false,
                  "description": "Sharwama contains no animal products, perfect for dieters",
                  "id": "36d525d1-efc9-4b75-9999-3e3d8dc64ce3",
                  "imageURL": "https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&h=350",
                  "price": "1200.00",
                  "quantity": 1,
                  "title": "Vegetable Sharwama and Guava Smoothie",
                  "vegetarian": true,
                },
              ],
              "status": "canceled",
              "updatedAt": "2018-05-27T00:47:03.687Z",
            },
          ]
        }
      />
    </div>
  </React.Fragment>
</CatererView>
`;
